documentation/modules/deploying/proc-deploy-topic-operator-standalone.adoc

```diff
@@ -144,6 +144,28 @@ env:
 <3> The keystore contains the private key for mTLS authentication.
 <4> The password for accessing the keystore.
 
+. If you need a custom `SASL` configuration. For example, to access a cluster in a cloud provider with a custom login module, the `custom` `SASL_MECHANISM` allows for `SASL_CUSTOM_CONFIG`. +
```

Comment:
I expected that if you need AWS IAM or AZURE, etc. you would create an image with those so we don't have to include every possible jar. I would like an standard mechanism for external jars other than a new image, but that may be a different conversation.

Reply:
Sorry, I don't know what you mean by BTO?

Reply:
> I would like an standard mechanism for external jars other than a new image, but that may be a different conversation.

As you say, we can discuss that in a separate thread, but for now, we should include a Dockerfile example for this specific use case, so that other users can easily adapt to their needs.

> Sorry, I don't know what you mean by BTO?

Bidirectional Topic Operator, it's the old deprecated implementation. In the latest Strimzi release you have the UTO (Unidirectional Topic Operator) enabled by default. https://strimzi.io/blog/2023/11/02/unidirectional-topic-operator

Reply:
Roger that, I get that in Dockerfile example.

Ah, I see, I'll get on that.

Reply:
I've run into a problem on MSK Serverless. BatchingTopicController is trying to find out if auto create topics is on. So a pretty reasonable question, but MSK Serverless doesn't expose that info through the usual Kafka mechanism. :/

**Does it even make sense to find pluggable solution for this?**

`Exception in thread "main" java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.InvalidRequestException: Amazon MSK Serverless doesn't support fetching broker or broker-logger configurations.
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
        at org.apache.kafka.common.internals.KafkaFutureImpl.get(KafkaFutureImpl.java:165)
        at io.strimzi.operator.topic.v2.BatchingTopicController.<init>(BatchingTopicController.java:108)
        at io.strimzi.operator.topic.v2.TopicOperatorMain.<init>(TopicOperatorMain.java:70)
        at io.strimzi.operator.topic.v2.TopicOperatorMain.operator(TopicOperatorMain.java:152)
        at io.strimzi.operator.topic.v2.TopicOperatorMain.main(TopicOperatorMain.java:146)
Caused by: org.apache.kafka.common.errors.InvalidRequestException: Amazon MSK Serverless doesn't support fetching broker or broker-logger configurations.`

Reply:
> I expected that if you need AWS IAM or AZURE, etc. you would create an image with those so we don't have to include every possible jar.

You should explain this in the documentation. Right now someone reading it might not guess all the steps you're expecting them to make.

Reply:
> `org.apache.kafka.common.errors.InvalidRequestException: Amazon MSK Serverless doesn't support fetching broker or broker-logger configurations`

Perhaps one way to handle this would be to allow the skipping of this check with an env var, e.g. `ASSUME_AUTO_CREATE_TOPICS=FALSE`. If this was not set we'd do the check which we currently do. If it was `=FALSE` then we just trust that the person configuring the operator is correct and skip the check. And in the unlikely case where it's `=TRUE` we fail in the same way as if we'd done the check ourselves.

Reply:
Sounds like a plan if you all are good with it. Should I proceed as such?

Reply:
Just a note. It works perfectly for normal "Provisioned" MSK clusters using IAM.

Reply:
On another note, the details of a working example might be a lot for the docs. What do you all think of a separate doc to explain the AWS example with a link in the Standalone doc?

Reply:
This is a repo I did to show the example. [MSK-IAM-Example](https://github.com/john-mcpeek/msk-iam-example/tree/master)

Reply:
I think a generic Dockerfile example with <placeholders> is enough. As mentioned before, we should also update the BTO documentation. Alternatively, we can only implement and document this feature for the UTO, as the BTO is going away in two releases.

Reply:
Actually, BTO is scheduled for removal in 0.41. So unless we change it it will be gone in few weeks.

Reply:
BTO is gone in the meantime, so less work to do :)